src/aics-image-viewer/state/subscribers.ts

```diff
@@ -0,0 +1,179 @@
+import { colorArrayToFloats } from "../shared/utils/colorRepresentations";
+import {
+  alphaSliderToImageValue,
+  brightnessSliderToImageValue,
+  densitySliderToImageValue,
+  gammaSliderToImageValues,
+} from "../shared/utils/sliderValuesToImageValues";
+import { select, type useViewerState, type ViewerStore } from "./store";
+
+const REF_EQ = { fireImmediately: true };
+const DEEP_EQ = { fireImmediately: true, equalityFn: shallow };
+
+export const subscribeViewToState = (store: typeof useViewerState, view3d: View3d): (() => void) => {
+  const unsubscribers = [
+    // view mode
+    store.subscribe(
+      select("viewMode"),
+      (viewMode) => {
+        view3d.setCameraMode(viewMode);
+        view3d.resize(null);
+      },
+      REF_EQ
+    ),
+
+    // camera state
+    store.subscribe(
+      select("cameraState"),
+      (cameraState) => {
+        if (cameraState) {
+          view3d.setCameraState(cameraState);
+        }
+      },
+      DEEP_EQ
+    ),
+
+    // autorotate
+    store.subscribe(select("autorotate"), view3d.setAutoRotate.bind(view3d), REF_EQ),
+
+    // show axes
+    store.subscribe(select("showAxes"), view3d.setShowAxis.bind(view3d), REF_EQ),
+
+    // background color
+    store.subscribe(
+      select("backgroundColor"),
+      (backgroundColor) => view3d.setBackgroundColor(colorArrayToFloats(backgroundColor)),
+      REF_EQ
+    ),
+
+    // brightness
+    store.subscribe(
+      select("brightness"),
+      (brightness) => {
+        view3d.updateExposure(brightnessSliderToImageValue(brightness));
+      },
+      REF_EQ
+    ),
+  ];
+
+  return () => unsubscribers.forEach((unsubscribe) => unsubscribe());
+};
+
+type AxisClipUpdateInfo = {
+  region: [number, number];
+  slice: number;
+  viewMode: ViewMode;
+};
+
+const selectAxisClipUpdateInfo = (axis: AxisName): ((store: ViewerStore) => AxisClipUpdateInfo) => {
+  return ({ region, slice, viewMode }) => ({ region: region[axis], slice: slice[axis], viewMode });
+};
+
+export const subscribeImageToState = (store: typeof useViewerState, view3d: View3d, image: Volume): (() => void) => {
+  const axisClipUpdater = (axis: AxisName) => {
+    return ({ region: [minval, maxval], slice, viewMode }: AxisClipUpdateInfo) => {
+      // Logic to determine axis clipping range, for each of x,y,z,3d slider:
+      // if slider was same as active axis view mode:  [viewerSettings.slice[axis], viewerSettings.slice[axis] + 1.0/volumeSize[axis]]
+      // if in 3d mode: viewerSettings.region[axis]
+      // else: [0,1]
+      let isOrthoAxis = false;
+      let axismin = 0.0;
+      let axismax = 1.0;
+      if (viewMode === ViewMode.threeD) {
+        axismin = minval;
+        axismax = maxval;
+      } else {
+        isOrthoAxis = activeAxisMap[viewMode] === axis;
+        const oneSlice = 1 / image.imageInfo.volumeSize[axis];
+        axismin = isOrthoAxis ? slice : 0.0;
+        axismax = isOrthoAxis ? slice + oneSlice : 1.0;
```

Comment:
Maybe explicitly note here that, in orthographic slice mode, the region is reset to the [0, 1] range for all axes other than the one currently being viewed. The comment on L80-L83 could maybe be clarified too?